app/controllers/api_controller.rb

```diff
@@ -1,9 +1,23 @@
 # frozen_string_literal: true
 
 class ApiController < ApplicationController
+  include Pundit
   include DeviseTokenAuth::Concerns::SetUserByToken
 
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
   protect_from_forgery with: :null_session
 
   before_action :authenticate_user!
+  before_action :verify_user
+
+  private
+
+  def verify_user
+    authorize User.find(params[:user_id]) if params[:user_id].present?
```

Comment:
I don't fully get this 🤔 Why are you calling `authorize` here? To which action do you want to check authorization for?

Reply:
My idea was to limit the actions to the `current_user`. There are routes where the `user` includes in its parameters

Reply:
Good! I assume your concern is about this action https://github.com/wolox-training/sa-rails/blob/f65b5c72b8ecdc0baff3d1af5b36177f57d05c21/app/controllers/rents_controller.rb#L8 In this case, instead of using `params[:user_id]` you could use `current_user`, which will return an instance of the logged in user.